server/delay_server/src/irodsDelayServer.cpp

```diff
@@ -676,6 +677,51 @@ namespace
     } // make_delay_queue_query_processor
 } // anonymous namespace
 
+bool is_current_server_leader(void) {
+    std::string hostname;
+    char hn[HOST_NAME_MAX + 1]{};
+
+    if (gethostname(hn, sizeof(hn)) != 0) {
+        logger::delay_server::warn("is_current_server_leader: Failed to retrieve local server's hostname.");
+        return false;
+    }
+
+    if (const auto hn_resolved = resolve_hostname(hn, hostname_resolution_scheme::match_preferred); hn_resolved) {
+        hostname = *hn_resolved;
+    }
+    else {
+        hostname = std::string{hn};
+    }
+    auto conn = get_new_connection(std::nullopt);
+    std::string result;
+    {
```

Comment:
How about we drop the additional scope (i.e. remove curly braces) and move `std::string result;` to where it's used.

The additional scoping isn't really necessary here.

Perhaps we don't even need `result` ...

Reply:
Lifted from rodsServer as well, but I agree; should be fine for it to be freed end-of-function instead of this scope. Fixing!